deployments/optimism-goerli/usdc/migrations/1679592519_configurate_and_ens.ts

```diff
@@ -0,0 +1,202 @@
+  prepare: async (deploymentManager: DeploymentManager) => {
+    return {};
+  },
+
+  enact: async (deploymentManager: DeploymentManager, govDeploymentManager: DeploymentManager) => {
+    const trace = deploymentManager.tracer();
+    const ethers = deploymentManager.hre.ethers;
+    const { utils } = ethers;
+
+    const {
+      bridgeReceiver,
+      comet,
+      cometAdmin,
+      configurator,
+      rewards,
+      USDC: optimismUSDC,
+      COMP: optimismCOMP
+    } = await deploymentManager.getContracts();
+
+    const {
+      optimismL1CrossDomainMessenger,
+      optimismL1StandardBridge,
+      governor,
+      USDC: goerliUSDC,
+      COMP: goerliCOMP,
+    } = await govDeploymentManager.getContracts();
+
+    // ENS Setup
+    // See also: https://docs.ens.domains/contract-api-reference/name-processing
+    const ENSResolver = await govDeploymentManager.existing('ENSResolver', ENSResolverAddress, 'goerli');
+    const subdomainHash = ethers.utils.namehash(ENSSubdomain);
+    const optimismGoerliChainId = (await deploymentManager.hre.ethers.provider.getNetwork()).chainId.toString();
+    const newMarketObject = { baseSymbol: 'USDC', cometAddress: comet.address };
+    const officialMarketsJSON = JSON.parse(await ENSResolver.text(subdomainHash, ENSTextRecordKey));
+    if (officialMarketsJSON[optimismGoerliChainId]) {
+      officialMarketsJSON[optimismGoerliChainId].push(newMarketObject);
+    } else {
+      officialMarketsJSON[optimismGoerliChainId] = [newMarketObject];
+    }
+
+    const configuration = await getConfigurationStruct(deploymentManager);
+
+    const setConfigurationCalldata = await calldata(
+      configurator.populateTransaction.setConfiguration(comet.address, configuration)
+    );
+    const deployAndUpgradeToCalldata = utils.defaultAbiCoder.encode(
+      ['address', 'address'],
+      [configurator.address, comet.address]
+    );
+    const l2ProposalData = utils.defaultAbiCoder.encode(
+      ['address[]', 'uint256[]', 'string[]', 'bytes[]'],
+      [
+        [configurator.address, cometAdmin.address],
+        [0, 0],
+        [
+          'setConfiguration(address,(address,address,address,address,address,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint104,uint104,uint104,(address,address,uint8,uint64,uint64,uint64,uint128)[]))',
+          'deployAndUpgradeTo(address,address)'
+        ],
+        [setConfigurationCalldata, deployAndUpgradeToCalldata]
+      ]
+    );
+
+    const USDCAmountToBridge = exp(10, 6);
+    const COMPAmountToBridge = exp(10_000, 18);
+
+    const goerliActions = [
+      // 1. Set Comet configuration and deployAndUpgradeTo new Comet on Optimism-Goerli.
+      {
+        contract: optimismL1CrossDomainMessenger,
+        signature: 'sendMessage(address,bytes,uint32)',
+        args: [bridgeReceiver.address, l2ProposalData, 5_000_000] // XXX find a reliable way to estimate the gasLimit
```

Comment:
Working on this right now.

Reply:
I haven't been able to get an accurate estimate of the L2 gas, as the estimate is off by almost 1mn gas. I know🥈 has been hitting a similar issue when estimating gas for Arbitrum.

I wonder if it's because I'm currently running these estimates in scenarios against a Hardhat node. I'll try again once the contracts have been deployed on-chain so I can run the estimate against an actual node.

Reply:
For posterity: this appears to be a known limitation when using https://github.com/ethereum-optimism/optimism/blob/develop/packages/sdk/src/cross-chain-messenger.ts which may require a contract upgrade to fix on their end. The current recommendation is to use a hardcoded limit